test/test.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const assert = require('assert');
+const spawn = require('child_process').spawn;
+const exec = require('child_process').exec;
+const request = require('request');
+const util = require('util');
+
+describe('nodejs-docker', () => {
+
+  it('serves traffic on 8080', (done) => {
+    runDocker('test/express', 8080, (host, callback) => {
+      setTimeout(() => {
+        request(`http://${host}:8080`, (err, res, body) => {
+          if (err) {
+            console.error(`Error requesting content: ${util.inspect(err)}`);
+            throw err;
+          }
+          assert.equal(body, 'Hello World!');
+          callback(() => {
+            done();
+          });
+        });
+      }, 3000);
+    });
+  });
+
+  it('can use a custom nodejs version', (done) => {
+    runDocker('test/nodeversion', 3000, (host, callback) => {
+      setTimeout(() => {
+        request(`http://${host}:3000`, (err, res, body) => {
+          if (err) {
+            console.error(`Error requesting content: ${util.inspect(err)}`);
+            throw err;
+          }
+          assert.equal(body, 'v5.9.0');
+          callback(() => {
+            done();
+          });
+        });
+      }, 3000);
+    });
+  });
+
+});
+
+/**
+ * Start a docker process for the given test
+ */
+function runDocker(tag, port, callback) {
+  let d = spawn('docker', [
```

Comment:
How about to name the docker run with `--name` option and use the name later when stopping?